article/app/services/dotcomponents/ArticlePicker.scala

```diff
@@ -122,13 +122,28 @@ object ArticlePicker {
     isSupported
   }
 
+  def dcrShouldRender(request: RequestHeader): Boolean = {
+    // A special exception to allow us to render with DCR even is the page is not supported
+    // Always render DCR for this one particular article
+    val forceDcr = request.path == "/info/2019/dec/08/migrating-to-react"
+
+    forceDcr
+  }
+
   def getTier(page: PageWithStoryPackage)(implicit request: RequestHeader): RenderType = {
 
     val whitelistFeatures = featureWhitelist(page, request)
     val isEnabled = conf.switches.Switches.DotcomRendering.isSwitchedOn
     val isCommercialBetaUser = ActiveExperiments.isParticipating(DotcomRenderingAdvertisements)
 
-    val tier = if ((dcrCouldRender(page, request) && isEnabled && isCommercialBetaUser && !request.forceDCROff) || request.forceDCR) RemoteRender else LocalRenderArticle
+    val tier = if (dcrShouldRender(request)) {
```

Comment:
Or maybe even pull this out to a method called something like `shouldForceDcr` which takes the request and returns a boolean - `true` if the `forceDcr` qs arg is set OR if we're a whitelisted page, otherwise `false`.

Reply:
@tjmw, as per our chat irl, I've refactored things to flatten out the structure. I also generally made things more readable with some renaming and comments as I think it's important this section of code is clear.

I stuck with the 'shoulds' and 'could' as it ended up reading better that way